addons/website_sale/models/payment.py

```diff
@@ -1,30 +1,29 @@
 # -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp import SUPERUSER_ID
-from openerp.osv import orm, fields
+from odoo import api, fields, models
 
 
-class PaymentTransaction(orm.Model):
+class PaymentTransaction(models.Model):
     _inherit = 'payment.transaction'
 
-    _columns = {
-        # link with the sale order
-        'sale_order_id': fields.many2one('sale.order', 'Sale Order'),
-    }
+    # link with the sale order
+    sale_order_id = fields.Many2one('sale.order', string='Sale Order')
 
-    def form_feedback(self, cr, uid, data, acquirer_name, context=None):
+    @api.model
+    def form_feedback(self, data, acquirer_name):
         """ Override to confirm the sale order, if defined, and if the transaction
         is done. """
         tx = None
-        res = super(PaymentTransaction, self).form_feedback(cr, uid, data, acquirer_name, context=context)
+        res = super(PaymentTransaction, self).form_feedback(data, acquirer_name)
 
         # fetch the tx, check its state, confirm the potential SO
         tx_find_method_name = '_%s_form_get_tx_from_data' % acquirer_name
         if hasattr(self, tx_find_method_name):
-            tx = getattr(self, tx_find_method_name)(cr, uid, data, context=context)
-        if tx and tx.state == 'done' and tx.acquirer_id.auto_confirm == 'at_pay_confirm' and tx.sale_order_id and tx.sale_order_id.state in ['draft', 'sent']:
-            self.pool['sale.order'].action_confirm(cr, SUPERUSER_ID, [tx.sale_order_id.id], context=dict(context, send_email=True))
-        elif tx and tx.state not in ['cancel', 'error'] and tx.sale_order_id and tx.sale_order_id.state in ['draft']:
-            self.pool['sale.order'].force_quotation_send(cr, SUPERUSER_ID, [tx.sale_order_id.id], context=context)
+            tx = getattr(self, tx_find_method_name)(data)
+        if tx and tx.state == 'done' and tx.acquirer_id.auto_confirm == 'at_pay_confirm' and tx.sale_order_id.state in ['draft', 'sent']:
+            tx.sale_order_id.with_context(send_email=True).action_confirm()
+        elif tx and tx.state not in ['cancel'] and tx.sale_order_id.state in ['draft']:
```

Comment:
@rga-odoo commit 84573fe50b7c064a4bd69b51ce800e3658f8ef5c is **not** present in migrated code, please check your rebase!